### PR TITLE
Performance Form data validation

### PR DESCRIPTION
**Pull request by meganniu**
Save button is only enabled if all fields of a Performance Form are filled

### CODE REVIEW

File: src/components/performances/PerformanceForm.jsx [hunk 1]
```diff
@@ -171,7 +181,7 @@ class PerformanceForm extends React.Component {
             <Button type="default" onClick={this.handleCancel}>
               {formType === 'edit' ? 'cancel' : 'discard'}
             </Button>
-            <Button disabled={false} type="primary" onClick={this.handleSubmit}>
+            <Button disabled={disabled} type="primary" onClick={this.handleSubmit}>
```
Comment (YangStephen) [inline src/components/performances/PerformanceForm.jsx]: let's reorder these props

File: src/components/performances/PerformanceForm.jsx [hunk 2]
```diff
@@ -79,6 +88,7 @@ class PerformanceForm extends React.Component {
   render() {
     const { classes, formType } = this.props;
     const {
+      disabled,
```
Comment (YangStephen) [inline src/components/performances/PerformanceForm.jsx]: order this alphabetically

File: src/components/performances/PerformanceForm.jsx [hunk 3]
```diff
@@ -19,20 +20,28 @@ class PerformanceForm extends React.Component {
     const { currentValues } = props;
 
     this.state = {
+      disabled: true,
```
[the remark this answers is not in the document]
Comment (YangStephen) [reply src/components/performances/PerformanceForm.jsx]: Also, could we name this something more specific. `disabledSubmit` or something? This seems like the entire pane should be disabled.

File: src/components/performances/PerformanceForm.jsx [hunk 4]
```diff
@@ -19,20 +20,28 @@ class PerformanceForm extends React.Component {
...
       performers: currentValues.performers || '',
       school: currentValues.school || '',
       size: currentValues.size || 0
     };
   }
 
+  // disable save button if not all input fields are filled
+  static getDerivedStateFromProps(props, state) {
+    const values = pick(state, 
+      ['academicLevel', 'choreographers', 'competitionLevel', 'danceEntry',
```
Comment (YangStephen) [inline src/components/performances/PerformanceForm.jsx]: For these inline arrays, let's extract this into another const.

File: src/components/performances/PerformanceForm.jsx [hunk 5]
```diff
@@ -19,20 +20,28 @@ class PerformanceForm extends React.Component {
...
       danceTitle: currentValues.danceTitle || '',
-      disabled: false,
       options: {},
       performers: currentValues.performers || '',
       school: currentValues.school || '',
       size: currentValues.size || 0
     };
   }
 
+  // disable save button if not all input fields are filled
```
Comment (YangStephen) [inline src/components/performances/PerformanceForm.jsx]: NIT: Proper capitalization in comments + a period at the end please!

File: src/components/performances/PerformanceForm.jsx [hunk 6]
```diff
@@ -19,20 +20,28 @@ class PerformanceForm extends React.Component {
...
+  static getDerivedStateFromProps(props, state) {
```
Comment (YangStephen) [inline src/components/performances/PerformanceForm.jsx]: I'll circle back on this in person but I'm not sure if `getDerivedStateFromProps` is the best place to do this. We're actually not deriving the state from the props but instead from the state only.

File: src/components/performances/PerformanceForm.jsx (hunk 1 above)
Comment (meganniu) [reply src/components/performances/PerformanceForm.jsx]: Reordered with `type` first

File: src/components/performances/PerformanceForm.jsx (hunk 2 above)
Comment (meganniu) [reply src/components/performances/PerformanceForm.jsx]: Done

File: src/components/performances/PerformanceForm.jsx (hunk 4 above)
Comment (meganniu) [reply src/components/performances/PerformanceForm.jsx]: Extracted into constant `fields`

File: src/components/performances/PerformanceForm.jsx (hunk 5 above)
Comment (meganniu) [reply src/components/performances/PerformanceForm.jsx]: Done

File: src/components/performances/PerformanceForm.jsx (hunk 6 above)
Comment (YangStephen) [reply src/components/performances/PerformanceForm.jsx]: Keeping based on existing precedence, moving this to componentDidUpdate will cause double renders.

**Pull request merged**